kubeconfig.go

```diff
@@ -83,38 +81,47 @@ func (k KubeConfig) NewG8sClientForApp(ctx context.Context, app v1alpha1.App) (v
 // the current cluster is returned.
 func (k KubeConfig) NewK8sClientForApp(ctx context.Context, app v1alpha1.App) (kubernetes.Interface, error) {
 	secretName := secretName(app)
+	secretNamespace := secretNamespace(app)
 
 	// KubeConfig is not configured so connect to current cluster.
 	if secretName == "" {
 		return k.k8sClient, nil
 	}
 
-	kubeConfig, err := k.getKubeConfigFromSecret(ctx, secretName, secretNamespace(app))
+	restConfig, err := k.NewRESTConfigForApp(ctx, secretName, secretNamespace)
 	if err != nil {
-		return nil, err
+		return nil, microerror.Mask(err)
 	}
 
-	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeConfig)
+	client, err := kubernetes.NewForConfig(restConfig)
 	if err != nil {
 		return nil, err
 	}
+	return client, nil
+}
+
+func (k KubeConfig) NewRESTConfigForApp(ctx context.Context, secretName, secretNamespace string) (*restclient.Config, error) {
```

Comment:
Why methods of this struct don't have pointer receivers?

Reply:
Does it necessary? I thought the pointer itself only need when we actually change the variable inside the instance.

Reply:
You are right, but what if you introduce some field you can't copy? This will become a breaking change w/o a reason.

I think having value receivers makes sense only for pure data structures so you can expose methods like `String` or `MarshalJSON `.